contracts/Staking.sol

```diff
@@ -519,33 +553,34 @@ contract Staking is Autopetrified, ERCStaking, ERCStakingHistory, IStakingLockin
         emit LockAmountChanged(_accountAddress, _lockManager, _amount, true);
     }
 
-    function _decreaseLockAmountUnsafe(address _accountAddress, address _lockManager, uint256 _amount) internal {
+    function _unlockUnsafe(address _accountAddress, address _lockManager, uint256 _amount) internal {
         Account storage account = accounts[_accountAddress];
         Lock storage lock = account.locks[_lockManager];
 
+        uint256 lockAmount = lock.amount;
+        require(lockAmount >= _amount, ERROR_NOT_ENOUGH_LOCK);
```

Comment:
I think this is being repeated in all the methods that call `_unlockUnsafe`. Additionally, as I mentioned for `_unsafeUnstake` I don't understand the rationale for calling this "unsafe"

Reply:
Only half of them, but yes it’s a mess, fixing it. :sweat_smile: